modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2056,6 +2056,7 @@ void ONNXImporter::parseGemm(LayerParams& layerParams, const opencv_onnx::NodePr
     }
 
     layerParams.set("bias_term", node_proto.input_size() == 3);
+    layerParams.set("is_matmul", true);
```

Comment:
> is_matmul

Do we really need this?
Where is it used?

Does `InnerProduct` without `is_matmul` make sense (used somewhere)?

Such layer parameters should be described in headers' documentation, because this behavior is not obvious.

Reply:
In the current implementation, `InnerProduct` and `is_matmul` are implemented using the same code. Previously, it used to distinguish the two by judging whether `InputB` is const or not, which could not distinguish the two in some cases, so the `is_matmul` parameter  was introduced.

Reply:
Why importer should know about that implementation details?

Interfaces should be kept clear and stay implementation agnostic.
Implementation should use inheritance instead of adding of new strange external parameters.

Reply:
Because the implementation can‘t tell whether the operator is a matrix product or an inner product, it needs to be informed by the importer. I haven't thought of a better way to distinguish between the two in the implementation part.

Reply:
> cannot tell whether the operator is matrix multiplication or inner product, it needs to be informed by the importer

It is done by name/type of the layer: `layerParams.type = "InnerProduct";`

Reply:
Thanks! This is really useful! I will remove this parameter later.

Reply:
> It is done by name/type of the layer: `layerParams.type = "InnerProduct";`

This solution can‘t be implemented to distinguish `MatMul`, `GEMM` and `InnerProduct`, because their `layerParams.type` are all `InnerProduct`. They all use the [`fully_connected_layer`](https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/layers/fully_connected_layer.cpp) to implement. I still have to specific a parameter in importer.

**MatMul** use `layerParams.type = "InnerProduct"`: 
https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/onnx/onnx_importer.cpp#L2063-L2067 

**GEMM** use `layerParams.type = "InnerProduct"`: 
https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/onnx/onnx_importer.cpp#L2001-L2004

Other operators:
https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/caffe/caffe_io.cpp#L1069-L1070

https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/darknet/darknet_io.cpp#L181-L185

https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/tensorflow/tf_importer.cpp#L1001

https://github.com/opencv/opencv/blob/606c80344a156630d8348df6b7357971ea3f58fe/modules/dnn/src/torch/torch_importer.cpp#L597